server/src/main/java/org/opensearch/identity/MyShiroModule.java

```diff
@@ -0,0 +1,118 @@
+package org.opensearch.identity;
+
+import org.apache.shiro.authc.AuthenticationException;
+import org.apache.shiro.authc.UsernamePasswordToken;
+import org.apache.shiro.authz.Permission;
+import org.apache.shiro.authz.permission.AllPermission;
+import org.apache.shiro.authz.permission.RolePermissionResolver;
+import org.apache.shiro.authz.permission.WildcardPermission;
+import org.apache.shiro.mgt.SecurityManager;
+import org.apache.shiro.SecurityUtils;
+import org.apache.shiro.realm.Realm;
+import org.apache.shiro.realm.SimpleAccountRealm;
+import org.apache.shiro.mgt.DefaultSecurityManager;
+import org.apache.shiro.subject.SimplePrincipalCollection;
+import org.apache.shiro.subject.Subject;
+import org.opensearch.common.Strings;
+
+import java.util.Base64;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+
+/**
+ * Prototyping use of identity management (authentication & authorization) implemented with Shiro
+ * See more at https://shiro.apache.org/
+ */
+public class MyShiroModule {
+
+    public MyShiroModule() {
+        final Realm realm = new MyRealm();
+        final SecurityManager securityManager = new DefaultSecurityManager(realm);
+        // Note; this sets up security for the JVM, if we are crossing a JVM boundary will need to look at how this is made available
+        SecurityUtils.setSecurityManager(securityManager);
+    }
+
+    /**
+     * Generates an internal subject to run requests in this context
+     */
+    public static Subject getSubjectOrInternal() {
+        final Subject existingSubject = SecurityUtils.getSubject();
+        if (existingSubject.isAuthenticated()) {
+            return existingSubject;
+        }
+
+        // Used for tracing where this function is used that is visible in the log output
+        final Exception e = new Exception();
+        final StackTraceElement current = e.getStackTrace()[1];
+        final String sourceAnnotation = current.getFileName() + "." + current.getMethodName() + "@" + current.getLineNumber();
+
+        final Subject internalSubject = new Subject.Builder().authenticated(true)
```

Comment:
for my knowledge, what is the thought process behind setting `authenticated` to true?

Reply:
I'm presuming this creates an "internal" user identity (`Subject`) which allows internal transport communication generated by OpenSearch code itself to assume it's authenticated.

Reply:
I did not pull this design forward into the interfaces, we can consider adding it to Subject if it comes up